### DISCUSSION TITLE
Why do contact names have to be unique within a contact group?

### DISCUSSION THREAD

**Discussion by mwobst** [netbox-community/netbox#14071] [Q&A]
I – or rather we – do not see any real purpose/need in that restriction. If there is any, please enlighten me/us ;)

Right now we have the problem that using contact groups is tedious because some names have to be prefixed. Which is not exactly the nicest thing.

**Comment by wz4** [comment]
I have also encountered this issue and am curious as to why. Two contacts with the same name is not a unique scenario. I would also like to see max character length on some of the model fields increased, title and address specifically.

**Comment by ghost** [comment]
can you provide a specific example of duplicate contact names within a group?
you have 2 people named john smith who are in the same group?

**Reply by mwobst** [reply]
The netbox documentation provides not a reason why contact’s names must be unique within a group. I t doesn’t even state a specific purpose of these groups. So it is perfectly thinkable that there are two contacts with the same name.

Or in short: If there is no reason for a limitation, don’t impose a limit. If there is one, please tell, because then I can communicate it within my work place.

**Reply by wz4** [reply]
Yes, two contacts in the same group with the same name.

https://github.com/netbox-community/netbox/blob/develop/netbox/tenancy/models/contacts.py#L97

**Reply by ghost** [reply]
> The netbox documentation provides not a reason why contact’s names must be unique within a group. I t doesn’t even state a specific purpose of these groups. So it is perfectly thinkable that there are two contacts with the same name.
> 
> Or in short: If there is no reason for a limitation, don’t impose a limit. If there is one, please tell, because then I can communicate it within my work place.

I certainly understand that point of view, but I'm still not understanding your use case.
Can you provide a clear example of when you've run into this limitation? Do you really have 2 people named john smith in the same group?

btw, the documentation does in fact state that contact names must be unique within a group. it also lists what the purpose of contact groups is for.
https://netboxlabs.com/docs/netbox/en/stable/models/tenancy/contactgroup/

**Reply by wz4** [reply]
I’m automating the import of LDAP contacts to a group with that name. My workaround is to append an attribute to the name, which works fine in my case. The character limit on some fields is still a problem.

**Reply by Kiln707** [reply]
I see that this has been dead for several months, But I ran into the same issue.

Documentation states that:
"A contact represents an individual or group that has been associated with an object in NetBox for administrative reasons. For example, you might assign one or more operational contacts to each site."
and
"Contacts can be organized into arbitrary groups. These groups can be recursively nested for convenience. Each contact within a group must have a unique name, but other attributes can be repeated."

@goteamkor 
Documentation does state the limitation but does not explain Why the limitation exists.
Also, "Contacts are to be organized into arbitrary groups" is the purpose. So when we run into a situation where two people with the same name and our same arbitrary rule that places a contact into our arbitrary group, we suddenly cant. 

Why does this limitation on contact groups exist? I can have multiple contacts with the exact same data (Exception is ID and timestamps)
![image](https://github.com/user-attachments/assets/750547bd-a8d5-49ea-9a38-72512d84e7b7)
![image](https://github.com/user-attachments/assets/46060f83-302b-4cb0-a44e-40ed01b32602)

This isn't a problem for contacts without a group or in different groups and Contact Groups are supposed to be arbitrary.

So in my case, I'm creating groups based on departments, **and yes**, I have two people with the same name, different titles, and in the same department. Pretty wild.

I created a custom field for uniquely tracking each contact but now I am unable to utilize my Contact Group as I designed it, and it seems to be an arbitrary limitation.

Is there a **Real** reason that the names **must** be unique? Or is this limitation artificial? Arbitrary means it isn't supposed to matter, its there for me to organize my contacts, but there is a rule preventing me from organizing my contacts in my arbitrary way.

That is what I am trying to understand. Either Arbitrary Groups doesn't really mean arbitrary and there is supposed to be a certain way I am supposed to use them due to the structure of DB or something, or the Unique Name rule is arbitrary and an artificial limitation on users and should be removed to allow Arbitrary use.

Any information on this would be appreciated.

Thanks!
